### PR TITLE
Add linter(eslint)

### PR DESCRIPTION
**Pull request by ashu8912**

### CODE REVIEW

**Comment by ashu8912** [comment]
fixes #149

File: frontend/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "parser": "babel-eslint",
+    "plugins": [
+      "react"
+    ],
+    "rules": {
+      "space-in-parens": [ 0, "always" ],
+      "template-curly-spacing": [ 2, "always" ],
+      "array-bracket-spacing": [ 2, "always" ],
+      "object-curly-spacing": [ 2, "always" ],
+      "computed-property-spacing": [ 2, "always" ],
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: I would have to test how these work against what we have. I think it's better to have these turned off/set to never.

File: frontend/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      "no-multiple-empty-lines": [ 2, { "max": 1, "maxEOF": 0, "maxBOF": 0 } ],
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: I want a newline at the end of files (so Github and other tools don't warn us). Do we need `maxEOF` to be 1?

File: frontend/.eslintrc (hunk 1 above)
Comment (joaquimrocha) [reply frontend/.eslintrc]: I mean it should be allowed, but not enforced at the moment. And we can change it later.

File: frontend/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      "quotes": [ 1, "single", "avoid-escape" ],
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: `avoid-escape` seems to be deprecated in favor or `avoidEscape`: https://eslint.org/docs/rules/quotes#top
We also want `allowTemplateLiterals`.

File: frontend/.eslintrc [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      "no-use-before-define": [ 2, { "functions": false } ],
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: I don't want to enforce this for now (nor to have a warning).

File: frontend/.eslintrc [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      "semi": [0, "never"],
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: I want us to have semicolons.

File: frontend/.eslintrc [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      "prefer-const": 1,
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: Cool.

File: frontend/.eslintrc [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+      "react/prefer-es6-class": 0,
+      "react/jsx-filename-extension": 0,
+      "react/jsx-curly-spacing": [ 2, "always" ],
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: I'm on the fence with this as I guess we'll have to change a lot of code. But as a rule I like it.

File: frontend/.eslintrc [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+      "react/jsx-indent": [ 2, 4 ],
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: Let's use 2-spaces as we've been doing.
Same for the Js code (maybe you need to add the rule).

File: frontend/.eslintrc [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+      "class-methods-use-this": [ 1 ],
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: Not really important IMO. So maybe let's not bother.

File: frontend/.eslintrc [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+      "no-case-declarations": [ 1 ],
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: :+1:

File: frontend/.eslintrc [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+      "no-return-assign": [ 1 ],
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: Let's go with error and always here.

File: frontend/.eslintrc [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+      "no-param-reassign": [ 1 ],
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: :+1:

File: frontend/.eslintrc [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+      "no-shadow": [ 1 ],
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: :+1:

File: frontend/.eslintrc [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+      "no-underscore-dangle" : [0, "always"]
```
Comment (joaquimrocha) [inline frontend/.eslintrc]: :+1:

File: frontend/package.json [hunk 15]
```diff
@@ -44,17 +44,21 @@
     "@iconify/react": "^1.1.1",
     "@iconify/tools": "^1.3.9",
     "babel-loader": "^8.0.6",
+    "customize-cra": "^0.9.1",
```
Comment (joaquimrocha) [inline frontend/package.json]: Why is this needed?
